src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamCredentialCacheTest.cs

```diff
@@ -47,8 +49,18 @@ public async Task SslStream_SameCertUsedForClientAndServer_Ok()
                     // by the server using certificates from the Trusted Root Authorities certificate store.
                     // The client side will use the Trusted Issuers List, if not empty, to filter proposed certificates.
 
-                    Assert.True(client.IsMutuallyAuthenticated);
-                    Assert.True(server.IsMutuallyAuthenticated);
+                    if (clientCertificateRequired)
```

Comment:
May be worth of finding one more test where callback is used and add similar check for completeness.

Reply:
Done, I created separate tests in separate file